base/array.jl

```diff
@@ -391,6 +391,22 @@ end
 one(x::AbstractMatrix{T}) where {T} = _one(one(T), x)
 oneunit(x::AbstractMatrix{T}) where {T} = _one(oneunit(T), x)
 
+# Constructors for nothing and missing
```

Comment:
Unless I am missing something this whole block can be replaced by
```julia
# type and dimensionality specified
Array{T,N}(::Void, d...) where {T,N} = fill!(Array{T,N}(uninitialized, d...), nothing)
Array{T,N}(::Missing, d...) where {T,N} = fill!(Array{T,N}(uninitialized, d...), missing)
# type but not dimensionality specified
Array{T}(::Void, d...) where {T} = fill!(Array{T}(uninitialized, d...), nothing)
Array{T}(::Missing, d...) where {T} = fill!(Array{T}(uninitialized, d...), missing)
```
and just let the `Array{T[,N]}(::Uninitialized, d...)` constructor do the conversion etc of the input args.

Reply:
Yes, it could, but I followed the pattern used by `uninitialized` constructors in sysimg.jl. I'm not sure how much it matters, but I think the idea of the call to `convert` is to avoid compiling a specialized function for each integer type. So I'd rather keep that even if that's more verbose unless we're completely sure it's not needed.

Reply:
I don't understand, you introduce these constructors for all different types though? Wouldn't my suggestion just mean that we instead use another specialized `Array{T}(uninitialized, ...)` constructor?

Reply:
Yeah, but you still need to compile the method itself, the `Array{T}(uninitialized, ...)` method for `Integer`, and then the `Array{T}(uninitialized, ...)` method for `Int`. By doing the conversion directly, you can avoid one intermediate step. At least, that's how I understand it.

Reply:
Yea, I just assumed that the `Array{T}(uninitialized, ...)` method for `Integer` was already compiled.

Reply:
I would err to code simplicity and clarity rather than worrying overmuch about compiling an additional method that likely will be compiled anyway :).